pkg/cache/v3/delta.go

```diff
@@ -16,73 +16,292 @@ package cache
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"sync/atomic"
 
+	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
 	"github.com/envoyproxy/go-control-plane/pkg/cache/types"
-	"github.com/envoyproxy/go-control-plane/pkg/server/stream/v3"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
 )
 
+type resourceWithNameAndVersion interface {
```

Comment:
The idea is to avoid constraints on the resources that are actually not needed. Any added resource version in the cache is already set with a name
For the version it avoids having to rehash the resource. In the future we could also avoid the double serialization we currently do, which is quite expensive in CPU